src/osd/OSDMap.cc

```diff
@@ -7039,7 +7039,7 @@ void OSDMap::check_health(CephContext *cct,
     }
     if (!detail.empty()) {
       ostringstream ss;
-      ss << detail.size() << " OSDs or CRUSH {nodes, device-classes} have {NOUP,NODOWN,NOIN,NOOUT} flags set";
+      ss << detail.size() << " OSDs or CRUSH {nodes, device-classes} have {NOUP|NODOWN|NOIN|NOOUT} flags set";
```

Comment:
Agreed, the commas are confusing to say the least. Technically I'm fine with the patch but it needs to be properly communicated (CDM + mail to ceph-users?) IMHO.

Reply:
Mmm, I think we can go even more explicit:

```
" There are OSDs or other crush CRUSH nodes have at least one of NOUP, NODOWN, NOIN, or NOOUT flags set"

Reply:
> 

+1